src/ubx_message.toit

```diff
@@ -0,0 +1,559 @@
+  message_id -> int:
+    return LITTLE_ENDIAN.uint8 payload 1
+
+  id_string_ -> string:
+    return "NAK"
+
+/**
+The UBX-CFG-MSG message.
+
+Configures the rate at which messages are sent by the receiver.
+*/
+class CfgMsg extends Message:
+  /** The UBX-CFG-MSG message ID. */
+  static ID ::= 0x01
+
+  /**
+  Constructs a coniguration message.
+
+  When sent to the receiver, the message with the given $msg_class and
+    $msg_id will be sent at the given $rate.
+  */
+  constructor.message_rate --msg_class --msg_id --rate:
+    super Message.CFG ID #[msg_class, msg_id, rate]
+
+  id_string_ -> string:
+    return "MSG"
+
+/**
+The UBX-CFG-RST message.
+
+Resets the receiver.
+*/
+class CfgRst extends Message:
+  /** The UBX-CFG-RST message ID. */
+  static ID ::= 0x04
+
+  /**
+  Constructs a reset message.
+
+  The default parameters are a controlled software reset with a cold start.
+  See the description for other parameter options.
+  */
+  constructor --clear_sections=0xFFFF --reset_mode=2:
+    super Message.CFG ID (ByteArray 4)
+    LITTLE_ENDIAN.put_uint16 payload 0 clear_sections
+    LITTLE_ENDIAN.put_uint8  payload 2 reset_mode
+    LITTLE_ENDIAN.put_uint8  payload 3 Message.RESERVED_
+
+  id_string_ -> string:
+    return "RST"
+
+/**
+The UBX-NAV-PVT message.
+
+Navigation, position, velocity, and time solution.
+*/
+class NavPvt extends Message:
+  /** The UBX-NAV-PVT message ID. */
+  static ID ::= 0x07
+
+  /** Unknown GNSS fix. */
+  static FIX_TYPE_UNKNOWN ::= 0
+  /** Dead reckoning only. */
+  static FIX_TYPE_DEAD ::= 1
+  /** 2D fix. */
+  static FIX_TYPE_2D ::= 2
+  /** 3D fix. */
+  static FIX_TYPE_3D ::= 3
+  /** GNSS and dead reckoning. */
+  static FIX_TYPE_GNNS_DEAD ::= 4
+  /** Time only fix. */
+  static FIX_TYPE_TIME_ONLY ::= 5
+
+  /** Constructs a poll UBX-NAV-PVT message. */
+  constructor.poll:
+    super Message.NAV ID #[]
+
+  id_string_ -> string:
+    return "PVT"
+
+  /** Whether the give $message is a UBX-NAV-PVT message. */
+  static is_instance message/Message -> bool:
+    return message.clazz == Message.NAV and message.id == ID
+
+  /** Whether this is a GNSS fix. */
+  is_gnss_fix -> bool:
+    return (flags & 0b00000001) != 0
+
+  /** The time un UTC. */
+  utc_time -> Time:
+    return Time.utc year month day hours minutes seconds --ns=nanoseconds
+
+  /** The GPS time of week of the navigation epoch. */
+  itow -> int:
+    assert: not payload.is_empty
+    return LITTLE_ENDIAN.uint32 payload 0
+  /** The year (UTC). */
+  year -> int:
+    assert: not payload.is_empty
+    return LITTLE_ENDIAN.uint16 payload 4
+  /**
+  The month (UTC).
+  In the range [1..12].
+  */
+  month -> int:
+    assert: not payload.is_empty
+    return LITTLE_ENDIAN.uint8 payload 6
+  /**
+  The day (UTC).
+  In the range [1..31].
+  */
+  day -> int:
+    assert: not payload.is_empty
+    return LITTLE_ENDIAN.uint8 payload 7
+  /**
+  The hours (UTC).
+  In the range [0..23].
+  */
+  hours -> int:
```

Comment:
Consider switching to `h` like we do in `Time`.
Same for `m`, `s` and `ns`.

That said: it looks like you are following naming-conventions from the UBX "datasheet", so changing the names is probably not a good idea.